### PR TITLE
Added identity op and a fix for LRN

### PR DESCRIPTION
**Pull request by vivek-rane**
This is branched off the current PR so please review starting with the last 3 commits.

### CODE REVIEW

**Comment by vrv** [comment]
(we'll review the other PR first, and then once that's in, we can review this one with just the diffs).

**Comment by vivek-rane** [comment]
Please proceed with the review :)

File: tensorflow/core/graph/mkl_layout_pass.cc [hunk 1]
```diff
@@ -1275,6 +1281,16 @@ void MklLayoutRewritePass::CopyAttrsBiasAddGrad(const Node* orig_node,
   nb->Attr("data_format", data_format);
 }
 
+void MklLayoutRewritePass::CopyAttrsIdentity(const Node* orig_node,
+                                           NodeBuilder* nb) {
```
Comment (zhangyaobit) [inline tensorflow/core/graph/mkl_layout_pass.cc]: Fix indentation.

File: tensorflow/core/kernels/mkl_lrn_op.cc [hunk 2]
```diff
@@ -533,22 +526,27 @@ class MklLRNGradOp : public OpKernel {
       }
     }
 
-    // convert input if needed
+        // convert input if needed
```
Comment (zhangyaobit) [inline tensorflow/core/kernels/mkl_lrn_op.cc]: Why added space?

File: tensorflow/core/util/mkl_util.h [hunk 3]
```diff
@@ -563,8 +563,9 @@ inline void CopyMklTensorInToOut(OpKernelContext* context, int idx_in,
   context->set_output(idx_meta_out, meta_output);
 }
 
-inline void CopyTFTensorInToOut(OpKernelContext* context, int idx_in,
-                                int idx_out, const TensorShape& shape) {
+inline void CopyTfTensorInToOutWithShape(OpKernelContext* context,
+                                int idx_in, int idx_out,
```
Comment (zhangyaobit) [inline tensorflow/core/util/mkl_util.h]: Fix indentation?

File: tensorflow/core/util/mkl_util.h [hunk 4]
```diff
@@ -580,6 +581,41 @@ inline void CopyTFTensorInToOut(OpKernelContext* context, int idx_in,
...
+  AllocateOutputSetMklShape(context, idx_out, mkl_shape_output);
+  if (IsRefType(context->input_dtype(idx_data_in))) {
+    context->forward_ref_input_to_ref_output(idx_data_in, idx_data_out);
+  } else {
+    context->set_output(idx_data_out, context->input(idx_data_in));
+  }
+}
+
+inline void ForwarMklTensorInToOut(OpKernelContext* context,
+                                      int idx_in, int idx_out) {
```
Comment (zhangyaobit) [inline tensorflow/core/util/mkl_util.h]: Fix indentation.

**Review by zhangyaobit — APPROVED** [review]
Looks good to me. Please fix some style issues.

**Comment by vrv** [comment]
btw our contribution page now has some helpful tools to deal with style issues: https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md#c-coding-style

File: tensorflow/core/util/mkl_util.h (hunk 3 above)
Comment (mdfaijul) [inline tensorflow/core/util/mkl_util.h]: Fixed indentation.

File: tensorflow/core/util/mkl_util.h (hunk 4 above)
Comment (mdfaijul) [inline tensorflow/core/util/mkl_util.h]: Fixed indentation

File: tensorflow/core/graph/mkl_layout_pass.cc (hunk 1 above)
Comment (mdfaijul) [inline tensorflow/core/graph/mkl_layout_pass.cc]: Fixed indentation.

**Comment by vivek-rane** [comment]
Style issues should be fixed now. I think we can proceed with unit tests.

**Comment by andydavis1** [comment]
@tensorflow-jenkins test this please

**Comment by vivek-rane** [comment]
👍

**Pull request merged**